### PR TITLE
Adding fuel core service monitor

### PR DESCRIPTION
**Pull request by rfuelsh**
We need to capture metrics in prometheus using fuel core metrics endpoint and hence we need a service monitor

### CODE REVIEW

File: deployment/monitoring/fuel-core-service-monitor.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+piVersion: monitoring.coreos.com/v1
```
Comment (Voxelot) [inline deployment/monitoring/fuel-core-service-monitor.yaml:1]:
```suggestion
apiVersion: monitoring.coreos.com/v1
```

Comment (rfuelsh) [reply deployment/monitoring/fuel-core-service-monitor.yaml]: nice catch

File: deployment/scripts/fuel-core-service-monitor-deploy.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    kube_prom_helm_chart_exists=$(helm list --namespace monitoring | grep kube-prometheus)
+    if [[ $? != 0 ]]; then
+        echo "kube-prometheus helm chart is not installed - please install it before redeploying this script ..."
+        exit 1;
+    elif [[ $kube_prom_helm_chart_exists ]]; then
+        echo "kube-prometheus helm chart is detected ..."
+    fi
+    
+    cd ../monitoring
```
[the remark this answers is not in the document]
Comment (rfuelsh) [reply deployment/scripts/fuel-core-service-monitor-deploy.sh:20]: https://github.com/FuelLabs/fuel-core/tree/add-fuel-core-service-monitor/deployment/monitoring -> created monitoring folder to add the service monitor yaml in

[the remark this answers is not in the document]
Comment (Voxelot) [reply deployment/scripts/fuel-core-service-monitor-deploy.sh:20]: ah sorry misread that

**Review by Voxelot — APPROVED** [review]

**Pull request merged**